guardian/facia-tool
# Detect invalid URL when creating plain snap link, and launch error modal

## What's changed?

This PR detects an invalid link when creating plain snap link, and launches an error modal.

An invalid link, for a snap link, is anything that isn't a valid URL, and doesn't start with a http(s) protocol (for example, a `mailto:` address would be pass the former test, but not the latter.)

Before (oh noo):
<img width="336" alt="Screenshot 2021-06-29 at 09 38 49" src="https://user-images.githubusercontent.com/7767575/123765644-d5a96880-d8bd-11eb-8a2f-6ecdd7318b4a.png">

After:
<img width="674" alt="Screenshot 2021-06-29 at 09 31 16" src="https://user-images.githubusercontent.com/7767575/123765665-d9d58600-d8bd-11eb-9d1a-64d99e3dae85.png">

## How to test

Drag and drop something egregious into the Fronts tool (e.g. this link, which you may need to save as a bookmarklet, and markdown refuses to turn into an `a` tag: `javascript:(function(){document.body.appendChild(document.createElement('script')).src='https://dashboard.ophan.co.uk/assets/js/heatmap-bookmarklet.js';})();`

You should get the above modal ... and definitely _not_ a snaplink.
